internal/client/statistic/monitor.go

```diff
@@ -8,43 +8,90 @@ import (
 	"time"
 
 	"github.com/ilya372317/must-have-metrics/internal/client/sender"
+	"github.com/ilya372317/must-have-metrics/internal/config"
 	"github.com/ilya372317/must-have-metrics/internal/dto"
+	"github.com/ilya372317/must-have-metrics/internal/logger"
 	"github.com/ilya372317/must-have-metrics/internal/server/entity"
 	"github.com/ilya372317/must-have-metrics/internal/utils"
+	"github.com/shirou/gopsutil/v3/cpu"
+	"github.com/shirou/gopsutil/v3/mem"
 )
 
 const counterName = "PollCount"
 const randomValueName = "RandomValue"
 const minRandomValue = 1
 const maxRandomValue = 50
+const chunkForRequestSize = 50
 
 type Monitor struct {
-	Data map[string]MonitorValue
+	Data         map[string]MonitorValue
+	ReportTaskCh chan func()
 	sync.Mutex
 }
 
-func New() Monitor {
-	return Monitor{Data: make(map[string]MonitorValue)}
+func New(rateLimit uint) *Monitor {
+	m := &Monitor{
+		Data:         make(map[string]MonitorValue),
+		ReportTaskCh: make(chan func(), rateLimit),
+	}
+	m.startWorkerPool(rateLimit)
+	return m
+}
+
+func (monitor *Monitor) startWorkerPool(rateLimit uint) {
+	for k := 0; k < int(rateLimit); k++ {
```

Comment:
rate limit это другое - это ограничение количества запросов в момент времени. Даже если по задумке эти значения совпадают, другой разработчик скажет что тут ошибка прочитав имя переменной
```suggestion
	for k := 0; k < int(poolSize); k++ {
```

Reply:
++